helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -151,7 +151,10 @@ public enum ClusterConfigProperty {
     HELIX_ENABLED_DISABLE_TIMESTAMP,
     HELIX_DISABLED_REASON,
     // disabled type should be a enum of org.apache.helix.constants.InstanceConstants.InstanceDisabledType
-    HELIX_DISABLED_TYPE
+    HELIX_DISABLED_TYPE,
+
+    // The last time when the on-demand rebalance is triggered.
+    LAST_ON_DEMAND_REBALANCE_TIMESTAMP
```

Comment:
Just curious, what happens in case of concurrent API calls? This field will be overwritten by the last caller. 

Also, do we need to retain the context specific to a partition or resource, or will we trigger cluster level rebalance (for-all)?